internal/controller/dragonfly_pod_lifecycle_controller.go

```diff
@@ -69,6 +72,30 @@ func (r *DfPodLifeCycleReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
+	// Get the role of the pod
+	role, roleExists := pod.Labels[resources.Role]
+	if !isPodReady {
+		restartCount := getRestartCount(pod)
+		if roleExists && role == "master" {
+			// If the master Pod is not ready and has restarted multiple times, initiate failover
+			if restartCount > 5 {
```

Comment:
hmm, 5 makes sense as when there is a problem with starting. 5 times is pretty quick.

Reply:
made the restartCount check `> 0` as there are two possibilites for this - 1) user created the Dragonfly resource but the master is not starting (i.e. keep crashing), in that case we switch the role to one of available replicas. (2) the running master Pod is restarted (e.g. when increasing pod cpu or memory limit/ or due to a bug), in that case it would initiate the failover to one of replicas.

The problem with `restartCount > 5` is that it won't trigger unless the restart count has crossed the "restartCount" limit.